util/versions/version.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package versions
+
+import (
+	semverv4 "github.com/blang/semver"
+)
+
+// GetHigherK8sVersion returns the higher k8s version out of a and b.
+func GetHigherK8sVersion(a, b string) string {
```

Comment:
WDYT about using this [compare function](https://pkg.go.dev/golang.org/x/mod/semver#Compare) instead? It seems like that would solve your issue of trying to handle error parsing.

Reply:
ack and makes sense, thank you for pointing me out to that function, will use this function which will  reduce a lot of chaos :smile:

Reply:
yup changed it accordingly.